Create ui-c8y-1019-22-0-private-smart-rule-not-need-datacontext-6031-graft-release-cd-6197.md (#1982)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1019-22-0-private-smart-rule-not-need-datacontext-6031-graft-release-cd-6197.md

---
date: ""

> title: private smart rule not need datacontext (#6031) [GRAFT][release/cd] (#6197)

Author: The Claude AI suggested to use the following title:
```suggestion
title: Private smart rules no longer require a data context
```

Reviewer: Already committed.

> title: private smart rule not need datacontext (#6031) [GRAFT][release/cd] (#6197)

Reviewer: ```suggestion
title: private smart rule no longer require contextData from c8y_Context property (#6031) [GRAFT][release/cd] (#6197)
```

> title: private smart rule no longer require contextData from c8y_Context property (#6031) [GRAFT][release/cd] (#6197)

Reviewer 2: ```suggestion
title: Private smart rules no longer require the contextData property
```

product_area: Application enablement & solutions
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-YdSEScrEC
    label: Cockpit
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-55916
version: 1019.22.0
---

> private smart rule not need datacontext (#6031) [GRAFT][release/cd] (#6197)

Author: The Claude AI suggested to use the following body:
```suggestion
In the past, private smart rules always required a data context to be specified, even if the rule did not actually use it. This unnecessary requirement has now been removed. Private smart rules can be created without specifying a data context, simplifying the rule creation process. This change only affects the creation of new private smart rules. Existing smart rules are not impacted.
```

> private smart rule not need datacontext (#6031) [GRAFT][release/cd] (#6197)

Reviewer: ```suggestion
In the past, private smart rules had property contextData in c8y_Context, even if the rule did not use it. This unnecessary requirement has now been removed. Private smart rules can be created without specifying a contextData, to avoid storing redundant MO data in the database. This change only affects the creation of new private smart rules. Existing smart rules are not impacted.
```

> In the past, private smart rules had property contextData in c8y_Context, even if the rule did not use it. This unnecessary requirement has now been removed. Private smart rules can be created without specifying a contextData, to avoid storing redundant MO data in the database. This change only affects the creation of new private smart rules. Existing smart rules are not impacted.

Reviewer 2: ```suggestion
In the past, private smart rules had the property `contextData` in `c8y_Context`, even if the rule did not use it. This behavior has changed. Private smart rules can now be created without the `contextData` property, to avoid storing redundant data in the database. This change only affects the creation of new private smart rules. Existing smart rules are not impacted.
```

In the past, private smart rules always required a data context to be specified, even if the rule did not actually use it. This unnecessary requirement has now been removed. Private smart rules can be created without specifying a data context, simplifying the rule creation process. This change only affects the creation of new private smart rules. Existing smart rules are not impacted.

Reviewer: I would not call it a "behaviour". 
It's more like about structure of the object:
```
{
	c8y_Context:
		contextData: {},
                someOtherProperty: "Hello World"
}
```
The final result of the change is that `contextData` is no longer part of `c8y_Context`.

Reviewer 2: @pmich-sag I rephrase it.
